### ISSUE TITLE
Record metadata on who learns commands

### ISSUE THREAD

**Issue by jakevoytko** [jakevoytko/crbot#12]
Predictably, my friends are adding weird commands, and changing commands in PMs, and not fessing up when we try to figure out whodunit. I'm probably doing this too.

Anyways, add `?meta <command>` to get metadata on who added a specific command, and implement command storage as a list of commands.

**Comment by drvan** [comment]
Does this affect #6 ? Particularly the key/value store interface - do you need more functionality of the datastore to be able to implement this?

**Comment by jakevoytko** [comment]
Initial thought: use Redis' lists, where the metadata is stored as serialized protobufs. So I'll likely add a second model implementation that abstracts this away; the current one I called `StringMap` (which is passed a bucket name on instantiation, and acts as a `map[string]string`), so it seems reasonable to have the command list one be a StringListMap (since it doesn't need to know that it's storing commands at the DB layer).

I ended up abstracting the bucket name away from the interface, which allows me to pass in individual `StringMap`s to features, which limits the scope of what they can access in Redis to exactly what they need.
